### ISSUE TITLE
"ITEM COLLISION true" error

### ISSUE THREAD

**Issue by perthblank** [BTMorton/angular2-grid#240]
Hi,

Today I suddenly find the library complains error once I drag the components. I'm running the demo and did not change anything.

the error is like
```
[1248] NgGrid.umd.min.js:1 ITEM COLLISION true
core.umd.min.js:31 EXCEPTION: Error in app/app.html:10:0 caused by: Maximum call stack size exceeded
...
```

Also there is 
```
[32] NgGrid.umd.min.js:1 LEFTOVER ITEM
```
error happens.

I'm using node 6.10.0, and errors occur on both angular 2.4.4 and 4.2.2 (my own app). Can someone give any suggestions?

Thanks in advance.

**Comment by Mani2693** [comment]
Issue arrived because of the fix #231 https://github.com/BTMorton/angular2-grid/issues/231#issuecomment-306123935
@BTMorton  Can you please confirm is it separate issue or issue because of #231  ?

Thanks
Mani

**Comment by BTMorton** [comment]
Not exactly, this issue came about as a result of an attempt at performance improvements in the grid and changes I made to the `_removeFromGrid` method. I think for now I'll have to put it back to how it was and give it another attempt later.
